### DISCUSSION TITLE
How do I set up Nav2?

### DISCUSSION THREAD

**Discussion by CFBerryhill** [iRobotEducation/create3_docs#602] [Questions] [answered]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.3

### Which RMW is your robot running?

I don't know

### Are there multiple Create 3 robots connected to your network?

None

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

None

### Describe your question.

I've downloaded Nav2 and am trying to run [this example](https://docs.nav2.org/tutorials/docs/navigation2_with_slam.html), but I am not sure how to configure Nav2 with the create3. I dont want to repeat a bunch of work because I know the odometry/tf frames and such are already set up. The page I linked does not make super clear what this `export TURTLEBOT3_MODEL=waffle` is meant to point at and how I may change that to be appriopriate for my own robot - or at least I havent found it. 

When I try to run the commands from the tutorial I get this in the output:

`[ERROR] [velocity_smoother-7]: process has died [pid 17079, exit code -9, cmd '/opt/ros/humble/lib/nav2_velocity_smoother/velocity_smoother --ros-args --log-level info --ros-args -r __node:=velocity_smoother --params-file /tmp/launch_params_inyf50xk -r /tf:=tf -r /tf_static:=tf_static -r cmd_vel:=cmd_vel_nav -r cmd_vel_smoothed:=cmd_vel'].
[ERROR] [waypoint_follower-6]: process has died [pid 17077, exit code -9, cmd '/opt/ros/humble/lib/nav2_waypoint_follower/waypoint_follower --ros-args --log-level info --ros-args -r __node:=waypoint_follower --params-file /tmp/launch_params_inyf50xk -r /tf:=tf -r /tf_static:=tf_static'].
[ERROR] [smoother_server-2]: process has died [pid 17069, exit code -9, cmd '/opt/ros/humble/lib/nav2_smoother/smoother_server --ros-args --log-level info --ros-args -r __node:=smoother_server --params-file /tmp/launch_params_inyf50xk -r /tf:=tf -r /tf_static:=tf_static'].
[ERROR] [behavior_server-4]: process has died [pid 17073, exit code -9, cmd '/opt/ros/humble/lib/nav2_behaviors/behavior_server --ros-args --log-level info --ros-args -r __node:=behavior_server --params-file /tmp/launch_params_inyf50xk -r /tf:=tf -r /tf_static:=tf_static'].
[ERROR] [lifecycle_manager-8]: process has died [pid 17081, exit code -9, cmd '/opt/ros/humble/lib/nav2_lifecycle_manager/lifecycle_manager --ros-args --log-level info --ros-args -r __node:=lifecycle_manager_navigation --params-file /tmp/launch_params_xqwq8v7e --params-file /tmp/launch_params_yctai18h --params-file /tmp/launch_params_iz1fhg88'].
[ERROR] [planner_server-3]: process has died [pid 17071, exit code -9, cmd '/opt/ros/humble/lib/nav2_planner/planner_server --ros-args --log-level info --ros-args -r __node:=planner_server --params-file /tmp/launch_params_inyf50xk -r /tf:=tf -r /tf_static:=tf_static'].`

It looks like there are some parameters or env variables not being set - but I am unsure of the proper way to set them or what to set them to.

I can run SLAM-TOOLBOX and all the create3 examples which I assume require much of the same configuration as nav2 - and have tried digging for more specific info on how people have set up their create3s with nav2 - [finding this page](https://github.com/iRobotEducation/create3_docs/discussions/68) that seems useful but I feel I dont have the context to really understand what is being said there.

Largely looking for pointers to specific docs I should be looking at or, if I'm missing a relatively simple step that too. Any help is appreciated, thanks!

**Comment by brianabouchard** [comment, marked as answer]
Hello! @kwujciak put together some instructions for mapping with the addition of Nav2 for use with the Create 3 Examples repo. Feel free to check them out [here](https://docs.google.com/document/d/1z7_fO3QsRu_dM8n0DV-_sXJ-JVsXuacByr7GLyxD87g/edit?usp=sharing).

**Reply by CFBerryhill** [reply]
Thanks for the tip, looks like I installed nav2 on the wrong computer! (Pi onboard the create instead of computer actually running rviz)  

This seems immedietly better, I see the tools in RViz and the topics on the SBC, but my robot still isn't moving. I suspect the problem is that the computer I'm running rviz on is up to jazzy, but the SBC is still on humble. Will try fixing that and will report back if that works.
